src/lib/operators/set_union.cpp

```diff
@@ -0,0 +1,371 @@
+#include "set_union.hpp"
+
+#include <algorithm>
+#include <chrono>
+#include <memory>
+#include <numeric>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "storage/chunk.hpp"
+#include "storage/reference_column.hpp"
+#include "storage/table.hpp"
+#include "types.hpp"
+
+/**
+ * ### SetUnion implementation
+ * The SetUnion Operator turns each input table into a ReferenceMatrix.
+ * The rows in the ReferenceMatrices need to be sorted in order for them to be merged, that is,
+ * performing the SetUnion  operation.
+ * Since sorting a multi-column matrix by rows would require a lot of value-copying, for each ReferenceMatrix, a
+ * VirtualPosList is created.
+ * Each element of this VirtualPosList references a row in a ReferenceMatrix by index. This way, if two values need to
+ * be swapped while sorting, only two indices need to swapped instead of a RowIDs for each column in the
+ * ReferenceMatrices.
+ * Using a implementation derived from std::set_union, the two virtual pos lists are merged into the result table.
+ *
+ *
+ * ### About ReferenceMatrices
```

Comment:
I feel like this should come before you talk about a `VirtualPosList`

Reply:
Want to explain the operator ("SetUnion implementation") before details ("About ReferenceMatrices"). I have added a reference to the paragraph that explains ReferenceMatrices to the place where they are mentioned first.